subsys/net/lib/lwm2m/lwm2m_obj_firmware.c

```diff
@@ -59,11 +60,85 @@ static struct lwm2m_engine_obj_inst inst;
 static struct lwm2m_engine_res_inst res[FIRMWARE_MAX_ID];
 
 static lwm2m_engine_set_data_cb_t write_cb;
+static lwm2m_engine_exec_cb_t update_cb;
 
 #ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
 extern int lwm2m_firmware_start_transfer(char *package_uri);
 #endif
 
+u8_t lwm2m_firmware_get_update_state(void)
+{
+	return update_state;
+}
+
+void lwm2m_firmware_set_update_state(u8_t state)
+{
+	/* Check LWM2M SPEC appendix E.6.1 */
+#if defined(CONFIG_ASSERT)
+	switch (state) {
+	case STATE_DOWNLOADING:
+		assert(update_state == STATE_IDLE);
```

Comment:
Are we sure that we want to assert here (even if CONFIG_ASSERT is enabled)?  That seems a bit harsh for passing in an invalid state.  Maybe ignore and print an error message?  Especially since this is exposed in application code.